src/auth/authclient.ts

```diff
@@ -13,31 +13,91 @@
 // limitations under the License.
 
 import {EventEmitter} from 'events';
-import {GaxiosOptions, GaxiosPromise, GaxiosResponse} from 'gaxios';
+import {Gaxios, GaxiosOptions, GaxiosPromise, GaxiosResponse} from 'gaxios';
 
 import {DefaultTransporter, Transporter} from '../transporters';
 import {Credentials} from './credentials';
 import {Headers} from './oauth2client';
+import {OriginalAndCamel, getOriginalOrCamel as getOpt} from '../util';
 
 /**
- * Defines the root interface for all clients that generate credentials
- * for calling Google APIs. All clients should implement this interface.
+ * Base auth configurations (e.g. from JWT or `.json` files)
  */
-export interface CredentialsClient {
+export interface AuthJSONOptions {
   /**
    * The project ID corresponding to the current credentials if available.
    */
-  projectId?: string | null;
+  project_id: string | null;
 
   /**
-   * The expiration threshold in milliseconds before forcing token refresh.
+   * The quota project ID. The quota project can be used by client libraries for the billing purpose.
+   * See {@link https://cloud.google.com/docs/quota Working with quotas}
    */
-  eagerRefreshThresholdMillis: number;
+  quota_project_id: string;
 
   /**
-   * Whether to force refresh on failure when making an authorization request.
+   * The default service domain for a given Cloud universe
    */
-  forceRefreshOnFailure: boolean;
+  universe_domain: string;
+}
+
+/**
+ * Base Auth Client configuration
+ */
+export type AuthClientOptions = Partial<
+  OriginalAndCamel<AuthJSONOptions> & {
```

Comment:
So this options object will end up having each of the AuthJSONOptions twice? Once with key in camelCase and one with the key in snake_case?

Reply:
Yep, with type hints and documentation. This way we can pass in the raw JSON properties from a file as-is or provide the options in a conventional JS pattern.